hunts/src/puzzle-table.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useTable, useExpanded } from "react-table";
+import { useGlobalFilter } from "./use-global-filter-fixed.js";
+import { matchSorter } from "match-sorter";
+
+export function PuzzleTable({ columns, data }) {
+    return usePuzzleTable({
+        columns: React.useMemo(() => columns, [columns]),
+        data: React.useMemo(() => data, [data]),
+    });
+}
+
+PuzzleTable.propTypes = {
+    columns: PropTypes.array.isRequired,
+    data: PropTypes.array.isRequired,
+};
+
+function textFilterFn(rows, id, filterValue) {
+    if (!filterValue || !filterValue.length) {
+        return rows;
+    }
+
+    const words = filterValue.split(' ');
+    if (!words) {
+        return rows;
+    }
+
+    const keys = ["values.name", "values.tags", "values.status", "values.answer"];
+    // Need to clone these results because this library is broken as shit
```

Comment:
match-sorter or react-table?

Reply:
react-table lol, I don't want to link to the issue since this is a public repo and I think github will notify them, but if you look at issue 2064 in the react-table github the filtering code seems kinda broken

Reply:
it's funny to me how much trouble this lib seems to be giving us, i see the guy shilling it on twitter non-stop 😆